src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -544,7 +549,18 @@ public void runRemotely(Step s) throws Throwable {
         conn.setRequestProperty("Content-Type", "application/octet-stream");
         conn.setDoOutput(true);
         Init2.writeSer(conn.getOutputStream(), Arrays.asList(token, s));
-        Throwable error = (Throwable) Init2.readSer(conn.getInputStream(), null);
+        Throwable error;
+        try {
+            error = (Throwable) Init2.readSer(conn.getInputStream(), null);
+        } catch (IOException e) {
```

Comment:
Note that if we keep the changes in `doStep` to run steps on a separate thread, I think this would really only happen if `doStep` itself had a problem. Still, it seems relatively harmless as a "just in case" kind of thing.